packages/wonder-blocks-dropdown/components/action-menu.test.js

```diff
@@ -80,4 +92,107 @@ describe("ActionMenu", () => {
+            >
+                <OptionItem label="Toggle A" value="toggle_a" />
+                <OptionItem label="Toggle B" value="toggle_b" />
+            </ActionMenu>,
+        );
+
+        menu.setState({open: true});
+
+        // toggle second OptionItem
+        const optionItem = menu.find(OptionItem).at(1);
+        optionItem.props().onToggle("toggle_b");
+
+        expect(selectedValues).toEqual(["toggle_a"]);
+    });
+
+    it("doesn't break with OptionItems but no onChange callback", () => {
+        const selectedValues = ["toggle_a", "toggle_b"];
+
+        const menu = mount(
+            <ActionMenu
+                menuText={"Action menu!"}
+                selectedValues={selectedValues}
+            >
+                <OptionItem label="Toggle A" value="toggle_a" />
+                <OptionItem label="Toggle B" value="toggle_b" />
+            </ActionMenu>,
+        );
+
+        menu.setState({open: true});
+
+        // toggle second OptionItem
+        const optionItem = menu.find(OptionItem).at(1);
+        optionItem.props().onToggle("toggle_b");
+
+        expect(selectedValues).toEqual(["toggle_a", "toggle_b"]);
+    });
+
+    it("works with extra selected values", () => {
+        let selectedValues = ["toggle_a", "toggle_z"];
+
+        const menu = mount(
+            <ActionMenu
+                menuText={"Action menu!"}
+                onChange={(values) => (selectedValues = values)}
+                selectedValues={selectedValues}
+            >
+                <OptionItem label="Toggle A" value="toggle_a" />
+                <OptionItem label="Toggle B" value="toggle_b" />
+            </ActionMenu>,
+        );
+
+        menu.setState({open: true});
+
+        // toggle second OptionItem
+        const optionItem = menu.find(OptionItem).at(0);
+        optionItem.props().onToggle("toggle_a");
+
+        expect(selectedValues).toEqual(["toggle_z"]);
+    });
+
+    it("can have a menu with a single item", () => {
+        const menu = mount(
+            <ActionMenu menuText={"Action menu!"}>
+                <ActionItem label="Action" />
+            </ActionMenu>,
+        );
+
+        menu.setState({open: true});
+
+        expect(menu.find(ActionItem)).toHaveLength(1);
+    });
+
+    it("can have a menu with no items", () => {
+        const menu = mount(<ActionMenu menuText={"Action menu!"} />);
+
+        menu.setState({open: true});
+
+        expect(menu.find(ActionItem)).toHaveLength(0);
+    });
+
+    it("can have falsy items", () => {
+        const showDeleteAction = false;
+        const menu = mount(
+            <ActionMenu menuText={"Action menu!"}>
+                <ActionItem label="Create" />
+                {showDeleteAction && <ActionItem label="Delete" />}
+            </ActionMenu>,
+        );
+
+        menu.setState({open: true});
+
+        expect(menu.find(ActionItem)).toHaveLength(1);
+        expect(menu.find(ActionItem).at(0)).toHaveProp("label", "Create");
```

Comment:
In terms of providing best practices around jest and enzyme, I think we should encourage people to use more specific matchers as opposed to:
```
expect(menu.find(ActionItem).length).toEqual(1);
expect(menu.find(ActionItem).at(0).props().label).toEqual("Create");
```

Reply:
https://www.npmjs.com/package/eslint-plugin-jest has a `prefer-to-have-length` rule that we could adopt.

Reply:
That makes sense to me :).